types/aws-lambda/trigger/codebuild-cloudwatch-state.d.ts

```diff
@@ -0,0 +1,110 @@
+import { Handler } from '../handler';
+
+export type CodeBuildCloudWatchStateHandler = Handler<CodeBuildCloudWatchStateEvent, void>;
+
+export type CodeBuildStateType = 'IN_PROGRESS' | 'SUCCEEDED' | 'FAILED' | 'STOPPED';
+export type CodeBuildPhaseType =
+    | 'COMPLETED'
+    | 'FINALIZING'
+    | 'UPLOAD_ARTIFACTS'
+    | 'POST_BUILD'
+    | 'BUILD'
+    | 'PRE_BUILD'
+    | 'INSTALL'
+    | 'QUEUED'
+    | 'DOWNLOAD_SOURCE'
+    | 'PROVISIONING'
+    | 'SUBMITTED';
+export type CodeBuildPhaseStatusType =
+    | 'TIMED_OUT'
+    | 'STOPPED'
+    | 'FAILED'
+    | 'SUCCEEDED'
+    | 'FAULT'
+    | 'CLIENT_ERROR';
+export type CodeBuildCacheType = 'NO_CACHE' | 'LOCAL' | 'S3';
+export type CodeBuildSourceLocationType =
+    | 'CODECOMMIT'
+    | 'CODEPIPELINE'
+    | 'GITHUB'
+    | 'GITHUB_ENTERPRISE'
+    | 'BITBUCKET'
+    | 'S3'
+    | 'NO_SOURCE';
+export type CodeBuildEnvironmentType =
+    | 'LINUX_CONTAINER'
+    | 'LINUX_GPU_CONTAINER'
+    | 'WINDOWS_CONTAINER'
+    | 'ARM_CONTAINER';
+export type CodeBuildEnvironmentPullCredentialsType = 'CODEBUILD' | 'SERVICE_ROLE';
+export type CodeBuildEnvironmentComputeType =
+    | 'BUILD_GENERAL1_SMALL'
+    | 'BUILD_GENERAL1_MEDIUM'
+    | 'BUILD_GENERAL1_LARGE'
+    | 'BUILD_GENERAL1_2XLARGE';
+export type CodeBuildEnvironmentVariableType = 'PARAMETER_STORE' | 'PLAINTEXT';
+
+export interface CodeBuildCloudWatchStateEvent {
```

Comment:
This should be able to use the `CloudWatchEvent<>` (?) template, passing in the detail type and detail.

Reply:
not sure what you mean here. Could you please explain a bit more? I based this PR on the `codepipeline-cloudwatch*` types. Can't seem to find if there

Reply:
Sorry, was me being on a phone again. I was referring to the `EventBridgeEvent<TDetailType extends string, TDetail>` in `trigger/eventbridge.d.ts` which seems to be what this structure is based on. It's more for users to be able to compose their own custom event structures though.